src/hotspot/share/oops/flatArrayKlass.hpp

```diff
@@ -41,14 +41,19 @@ class FlatArrayKlass : public ArrayKlass {
 
  private:
   // Constructor
-  FlatArrayKlass(Klass* element_klass, Symbol* name);
+  FlatArrayKlass(Klass* element_klass, Symbol* name, LayoutKind lk);
+
+  LayoutKind _layout_kind;
 
  public:
 
-  FlatArrayKlass() {}
+  FlatArrayKlass() {} // Why?
```

Comment:
Nuke it

Reply:
After investigation, it is used by CppVtableCloner<T>::initialize() to create the dummy instance. Comment has been updated.